.github/workflows/e2e.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         run: pnpm dlx concurrently --kill-others --success command-1 "$E2E_RUN_LOCAL_NET_CMD" 'pnpm --filter @mysten/sui test:e2e'
 
       - name: Run RPC/GrpahQL comaptability e2e tests
-        if: ${{ needs.diff.outputs.isGraphQlTransport == 'true' || needs.diff.outputs.isRust == 'true'}}
+        if: ${{ (needs.diff.outputs.isGraphQlTransport == 'true' || needs.diff.outputs.isRust == 'true') && github.ref == 'refs/heads/main' }}
```

Comment:
this will also disable this on all PRs right?

Reply:
we probably still want this for PRs, but just against localnet

Reply:
I pushed another one that checks specifically for devnet and testnet branches. I think that should work, what do you think? @hayes-mysten

Reply:
I am not very familiar with the github workflow syntax, but seems reasonable to me!